foundationdb/tests/conftest.py

```diff
@@ -1,12 +1,41 @@
 
+import json
+import time
+import os
 import pytest
 
+from datadog_checks.dev import WaitFor, docker_run, run_command
+
+from .common import E2E_INIT_CONFIG
+
+INSTANCE = {'use_sudo': False}
+CONFIG = {'init_config': E2E_INIT_CONFIG, 'instances': [INSTANCE]}
+HERE = os.path.dirname(os.path.abspath(__file__))
 
 @pytest.fixture(scope='session')
 def dd_environment():
-    yield
-
+    compose_file = os.path.join(HERE, 'docker', 'docker-compose.yaml')
+    with docker_run(compose_file=compose_file, conditions=[WaitFor(create_database)]):
+        yield CONFIG
 
 @pytest.fixture
 def instance():
-    return {}
+    return INSTANCE
+
+def create_database():
+    base_status = run_command('docker exec fdb-0 fdbcli --exec "status json"', capture=True, check=True)
+    status = json.loads(base_status.stdout)
+    if not status.get('client').get('database_status').get('available'):
+        run_command('docker exec fdb-0 fdbcli --exec "configure new single memory"', capture=True, check=True)
+
+    base_status = run_command('docker exec fdb-0 fdbcli --exec "status json"', capture=True, check=True)
+    status = json.loads(base_status.stdout)
+    i = 0
+    is_healthy = status.get('cluster').get('data').get('state').get('name') == 'healthy'
+    # Wait for 1 minute for the database to become available for testing
+    while i < 60 and not is_healthy:
+        i += 1
+        time.sleep(1)
+        base_status = run_command('docker exec fdb-0 fdbcli --exec "status json"', capture=True, check=True)
+        status = json.loads(base_status.stdout)
+        is_healthy = status.get('cluster').get('data').get('state').get('name') == 'healthy'
```

Comment:
Hm, this logic doesn't need to be duplicated both in and out of the loop, I think? It can just be in the loop and we have a (probably reasonable) 1 second wait before the first check?

Reply:
You're right, fixed.